src/main/java/com/gregtechceu/gtceu/api/item/ComponentItem.java

```diff
@@ -138,6 +146,9 @@ public InteractionResult useOn(UseOnContext context) {
                     return result;
                 }
             }
+            if (component instanceof IUIHolder<?>) {
+                PlayerInventoryUIFactory.INSTANCE.openFromHandClient(context.getHand());
+            }
```

Comment:
Do we want to do this here? if we do, don't we also want to overrride that in use(...) instead of only useOn(...)?